### PR TITLE
Fix: include root module in parameter calculation

### LINKED ISSUE #115: Passing a single layer (e.g., nn.Conv2d) resulted in 0 params
### Search before reporting

- [x] I have searched the appropriate Ultralytics [issues](https://github.com/ultralytics/) and found no similar bug report.

### Repository or project

https://github.com/ultralytics/thop

### Component / area

_No response_

### Bug description

- Got the zero parameters when profile a single layer (e.g. nn.Conv2d) 

### Environment

- OS: Windows11 24H2
- Python: 3.12.12
- Install: git

Package   Version
- filelock: 3.20.0
- fsspec: 2025.12.0
- Jinja2: 3.1.6
- MarkupSafe: 2.1.5
- mpmath: 1.3.0
- networkx: 3.6.1
- numpy: 2.3.5
- packaging: 25.0
- pillow: 12.0.0
- pip: 26.0.1
- setuptools: 80.10.2
- sympy: 1.14.0
- torch: 2.10.0+cu128
- torchvision: 0.25.0+cu128
- typing_extensions: 4.15.0
- wheel: 0.46.3

### Minimal reproducible example

```py
from thop import profile
import torch
import torch.nn as nn

class test_model(nn.Module):
    def __init__(self):
        super().__init__()
        self.layer = nn.Conv2d(3, 10, (3,3))

    def forward(self, input):
        return self.layer(input)


layer = nn.Conv2d(3, 10, (3,3))
model = test_model()

macs, weight = profile(layer, torch.zeros(1,3,224,224))
print("Single Layer Profile", macs, weight)
macs, weight = profile(model, torch.zeros(1,3,224,224))
print("Model Profile", macs, weight)
```

### Additional context

_No response_

### Are you willing to submit a PR?

- [x] Yes, I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by Yui-Arthur**
I have read the CLA Document and I sign the CLA

## Problem
When passing a single layer (e.g., `nn.Conv2d`) directly to `profile()`, the function reports 0 parameters. This happens because the parameter count of the root module itself was ignored during the dfs process.
(Fixes #115)

## Changes
- Initialized `total_params` with `module.total_params` in `dfs_count()` to ensure the root module's parameters are included.

## Minimal reproducible example (Issue #115)
```py
from thop import profile
import torch
import torch.nn as nn

class test_model(nn.Module):
    def __init__(self):
        super().__init__()
        self.layer = nn.Conv2d(3, 10, (3,3))

    def forward(self, input):
        return self.layer(input)


layer = nn.Conv2d(3, 10, (3,3))
model = test_model()

macs, weight = profile(layer, torch.zeros(1,3,224,224))
print("Single Layer Profile Weights: ", weight)
# Single Layer Profile Weights:  280.0
macs, weight = profile(model, torch.zeros(1,3,224,224))
print("Model Profile Weights: ", weight)
# Model Profile Weights:  280.0
```

Before fix the single layer profile will got zero weights, after fix the profile can get correct weights

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Fixes parameter counting in `thop/profile.py` by including the root module’s own parameters in the total. 🧮✅

### 📊 Key Changes
-Updates `dfs_count()` to initialize `total_params` from `module.total_params.item()` instead of `0`.
-Makes parameter aggregation consistent with operation counting (which already includes the current module). 🔧

### 🎯 Purpose & Impact
-Prevents undercounting parameters for models where the top-level/root module has parameters. 🐛➡️✅
-Improves accuracy and trustworthiness of profiling outputs (params + ops) for end users running THOP profiling. 📈

### CODE REVIEW

**Review by UltralyticsAssistant — DISMISSED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

PR looks clean. The change correctly aligns parameter counting with ops counting by including the root module’s parameters. No issues found.

**Review by UltralyticsAssistant — DISMISSED** [review]
## 🔍 PR Review 2

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Review complete. No issues found in the diff; the change correctly includes root module parameters without introducing problems.

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review 3

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Clean change that correctly includes root module parameters in the DFS count. No issues found.

**Comment by glenn-jocher** [comment]
Thanks for the fix @Yui-Arthur. I added regression coverage for profiling a root `nn.Conv2d` directly and for a root `nn.Sequential` wrapper, and adjusted parameter counting to use direct module parameters so container children are not double-counted. CI is passing and this is merged.
